### DISCUSSION TITLE
onMessage is not calling in ios platform.I haven't used notifee library.

### DISCUSSION THREAD

**Discussion by MarkJackson1997** [invertase/react-native-firebase#7628] [General]
onMessage is not calling in  iOS platform.I haven't used notifee library.
In android it is working fine when app is in foreground but in iOS side it is not calling in latest version.
Anything changed in latest version?

  **"react": "18.2.0",
    "react-native": "0.73.4",
    "@react-native-firebase/app": "^18.8.0",
    "@react-native-firebase/messaging": "^18.8.0",**
    
```
 useEffect(() => {
      const unsubscribeOnMessage = messaging().onMessage(async notification => {
      console.log("onMessage method")
      console.log("App in forground")

    const unsubscribeOnNotificationOpenedApp = messaging().onNotificationOpenedApp(remoteMessage => {
      console.log("onNotificationOpenedApp method")
      console.log("notification is open from background state")
      console.log("remote data" + JSON.stringify(remoteMessage))
   
    });

    return () => {
      unsubscribeOnMessage()
      unsubscribeOnNotificationOpenedApp()
    };
  }, [userToken])
  

```

  **notification payload:**
  {
    "to": "fcm token",
    "priority": "high",
    "notification": {
        "body": "notification body",
        "OrganizationId": "2",
        "content_available": true,
        "priority": "high",
        "title": "notification title"
       },
   "data" : {
 "priority" : "high",
 "body" : "New Announcement assigned",
  "title":"Title",
 "organization" :"Elementary school",
   "type":"notification"
}
}

**Comment by mikehardy** [comment]
No changes in this area I'm aware of. You might try it on a real device plugged in to development machine, with app either launched from Xcode so you may examine app log in Xcode console or with Console.app open and test device selected so you may watch device log

**Reply by MarkJackson1997** [reply]
Thanks for your response.
Currently have checked with iOS simulator as i have not real iOS device.
With simulator i observed onMessage methods log is not displaying when app is in foreground in iOS.Where same scenario is working in android as i am able to see logs in android.

Also notification is working fine when app is background and closed state in iOS.Only facing issue when app is foreground in iOS.In android it is working perfect for all 3 state(closed,background and foreground).

I will try to arrange iOS device and will let you know after checking in real device.
Thanks

**this is my payload to check notification.**
{
"to": "fcm token",
"priority": "high",
"notification": {
"body": "notification body",
"OrganizationId": "2",
"content_available": true,
"priority": "high",
"title": "notification title"
},
"data" : {
"priority" : "high",
"body" : "New Announcement assigned",
"title":"Title",
"organization" :"Elementary school",
"type":"notification"
}
}

**Comment by tiborazo** [comment]
Hi @MarkJackson1997 ,

Have you figured it out?

I am having the same issue. On Android everything works. On iOS background and killed state messages are also working, but onMessage event is never fired 😢 

I spent few weeks on this already, trying to change the push notifications package used by an old app to the latest React Native and Firebase messaging. Today I set up a new app from scratch - all new identifiers, bundle ids, certificates, etc. The only things installed in the new project are `yarn add @react-native-firebase/app` and `@react-native-firebase/messaging`. 

I followed all the latest guides on https://reactnative.dev/docs/environment-setup , https://rnfirebase.io/ , https://rnfirebase.io/messaging/usage and https://rnfirebase.io/messaging/usage/ios-setup.

I am sending the notifications with AWS SNS, but it is the same when sending from the Firebase Console.

"react": "18.2.0",
"react-native": "0.73.7",
"@react-native-firebase/app": "^19.2.2",
"@react-native-firebase/messaging": "^19.2.2"

I can provide more information if anyone can help, please.

Cheers,
Tibor Bicskei
